cdi-core/src/main/java/com/linkedin/cdi/util/AvroSchemaUtils.java

```diff
@@ -52,9 +52,13 @@ static List<String> getSchemaFieldNames(Schema schema) {
    * @return copy of schema field
    */
   static Schema.Field deepCopySchemaField(Schema.Field field) {
-      Schema.Field f = new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultVal(), field.order());
-      field.getProps().forEach(f::addProp);
-      return f;
+    Schema.Field f = new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultVal(), field.order());
+    field.getObjectProps().forEach(
+        (propName, propValue) -> {
+          if (propValue instanceof String) f.addProp(propName, (String) propValue);
```

Comment:
Why are we doing this only for String

Reply:
Previously we were using `field.getProps()` and the method only returns textual values. When bumping up the avro version to 1.9.2, the `getProps()` method was removed. At this time I am not able to determine whether we only need the Strings or we actually want to preserve all values, but in order to be consistent I decided to only retain Strings as we did earlier.
 /** Return the defined properties that have string values. */
  @Deprecated public Map<String,String> getProps() {
    Map<String,String> result = new LinkedHashMap<String,String>();
    for (Map.Entry<String,JsonNode> e : props.entrySet())
      if (e.getValue().isTextual())
        result.put(e.getKey(), e.getValue().getTextValue());
    return result;
  }